packages/trpc/server/routers/viewer/slots/reserveSlot.handler.ts

```diff
@@ -99,7 +100,18 @@ export const reserveSlotHandler = async ({ ctx, input }: ReserveSlotOptions) =>
       });
     }
   }
-  res?.setHeader("Set-Cookie", serialize("uid", uid, { path: "/", sameSite: "lax" }));
+  // We need this cookie to be accessible from embeds where the booking flow is displayed within an iframe on a different origin.
+  // For third‑party iframe contexts (embeds on other sites), browsers require SameSite=None and Secure to make the cookie available.
+  // For local development on http://localhost we fall back to SameSite=Lax to avoid requiring https during development.
+  const useSecureCookies = WEBAPP_URL?.startsWith("https://");
```

Comment:
I don't think WEBAPP_URL could be undefined
```suggestion
  const useSecureCookies = WEBAPP_URL.startsWith("https://");
```

Reply:
Dropped the optional chain on `WEBAPP_URL` — it always has a value so we can safely call `startsWith`.